wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -1647,17 +1651,25 @@ class KotlinGenerator private constructor(
           val fieldName = nameAllocator[field]
           val adapterName = field.getAdapterName()
 
-          if (field.type!!.isEnum) {
-            beginControlFlow("%L -> try", field.tag)
-            addStatement("%L", decodeAndAssign(field, fieldName, adapterName))
-            nextControlFlow("catch (e: %T)", ProtoAdapter.EnumConstantNotFoundException::class)
-            addStatement(
-              "reader.addUnknownField(%L, %T.VARINT, e.value.toLong())", tag,
-              FieldEncoding::class
-            )
-            endControlFlow()
-          } else {
-            addStatement("%L -> %L", field.tag, decodeAndAssign(field, fieldName, adapterName))
+          when {
+            field.type!!.isEnum -> {
+              beginControlFlow("%L -> try", field.tag)
+              addStatement("%L", decodeAndAssign(field, fieldName, adapterName))
+              nextControlFlow("catch (e: %T)", ProtoAdapter.EnumConstantNotFoundException::class)
+              addStatement(
+                "reader.addUnknownField(%L, %T.VARINT, e.value.toLong())", tag,
+                FieldEncoding::class
+              )
+              endControlFlow()
+            }
+            field.isPacked && field.isScalar -> {
```

Comment:
Today I learned enums can be packed!

Reply:
Yeah I was surprised, but I guess it makes sense